forecasting/src/autogluon/forecasting/trainer/auto_trainer.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from typing import Dict, Union, Optional, Any
+
+from ..models.presets import get_preset_models
+from .abstract_trainer import AbstractForecastingTrainer, TimeSeriesDataFrame
+
+logger = logging.getLogger(__name__)
+
+
+class AutoForecastingTrainer(AbstractForecastingTrainer):
+    def construct_model_templates(self, hyperparameters, **kwargs):
+        path = kwargs.pop("path", self.path)
+        eval_metric = kwargs.pop("eval_metric", self.eval_metric)
+        hyperparameter_tune = kwargs.get("hyperparameter_tune", False)
+        return get_preset_models(
+            path=path,
+            eval_metric=eval_metric,
+            prediction_length=self.prediction_length,
+            freq=self.freq,
+            hyperparameters=hyperparameters,
+            hyperparameter_tune=hyperparameter_tune,
+            quantiles=self.quantiles,
+            use_feat_static_cat=self.use_feat_static_cat,
+            use_feat_static_real=self.use_feat_static_real,
+            cardinality=self.cardinality,
+        )
```

Comment:
How do you ensure models don't have name collisions over multiple calls to `get_preset_models`?

The logic to ensure duplicate names aren't generated is implemented in AbstractModel currently, but it doesn't look like it is implemented here.

Imagine `self.model_info` already contains `model_name`. Must ensure `model_name` isn't used as a name for models in this call to avoid collision.

Ex in Tabular's AutoTrainer:

```
class AutoTrainer(AbstractTrainer):
    def construct_model_templates(self, hyperparameters, **kwargs):
        path = kwargs.pop('path', self.path)
        problem_type = kwargs.pop('problem_type', self.problem_type)
        eval_metric = kwargs.pop('eval_metric', self.eval_metric)
        quantile_levels = kwargs.pop('quantile_levels', self.quantile_levels)
        invalid_model_names = kwargs.pop('invalid_model_names', self._get_banned_model_names())
        silent = kwargs.pop('silent', self.verbosity < 3)
        ag_args_fit = kwargs.pop('ag_args_fit', None)
        if quantile_levels is not None:
            if ag_args_fit is None:
                ag_args_fit = dict()
            ag_args_fit = ag_args_fit.copy()
            ag_args_fit['quantile_levels'] = quantile_levels

        return get_preset_models(path=path,
                                 problem_type=problem_type,
                                 eval_metric=eval_metric,
                                 hyperparameters=hyperparameters,
                                 ag_args_fit=ag_args_fit,
                                 invalid_model_names=invalid_model_names,
                                 silent=silent, **kwargs)
```